core/merge.js

```diff
@@ -154,7 +154,9 @@ class Merge {
     }
     doc.sides[side] = ++rev
     doc._rev = info.rev
-    return this.pouch.put(doc)
+    const result = await this.pouch.put(doc)
+    delete doc._rev
```

Comment:
What's the point of removing the doc's revision here? We're not returning `doc` or reusing it so I assume we don't really care what its value is.

Reply:
Some unit tests failed when `_rev` is added to the doc.

Reply:
Wouldn't that be a problem with how the tests were written rather than the implementation in this case?

Reply:
OK. Another argument for a good refactoring of Merge :).